Event/KafkaEvent.php

```diff
@@ -32,4 +60,98 @@ public function getOrigin(): string
     {
         return $this->origin;
     }
+
+    /**
+     * Set error code
+     *
+     * @param int $errorCode
+     *
+     * @return KafkaEvent
+     */
+    public function setErrorCode(int $errorCode): KafkaEvent
+    {
+        $this->errorCode = $errorCode;
+
+        return $this;
+    }
+
+    /**
+     * Return error code
+     *
+     * @return int|null
+     */
+    public function getErrorCode()
+    {
+        return $this->errorCode;
+    }
+
+    /**
+     * @param string $reason
+     *
+     * @return KafkaEvent
+     */
+    public function setReason(string $reason): KafkaEvent
+    {
+        $this->reason = $reason;
+
+        return $this;
+    }
+
+    /**
+     * Return reason
+     *
+     * @return string|null
+     */
+    public function getReason()
+    {
+        return $this->reason;
+    }
+
+    /**
+     * @return float
+     */
+    public function getExecutionStart(): float
+    {
+        return $this->executionStart;
+    }
+
+    /**
+     * Set execution start of a request
+     *
+     * @return KafkaEvent
+     */
+    public function setExecutionStart(): KafkaEvent
+    {
+        $this->executionStart = microtime(true);
+
+        return $this;
+    }
+
+    /**
+     * @return KafkaEvent
+     */
+    public function setExecutionStop(): KafkaEvent
```

Comment:
There is a calcul, isn't set the execution time.
Maybe remove the set for executionstop